cmd/krew/cmd/install.go

```diff
@@ -90,45 +92,47 @@ Remarks:
 				return errors.New("--archive can be specified only with --manifest or --manifest-url")
 			}
 
-			var install []index.Plugin
+			var install []index.Receipt
 			for _, name := range pluginNames {
-				plugin, err := indexscanner.LoadPluginByName(paths.IndexPluginsPath(constants.DefaultIndexName), name)
+				indexName, pluginName := pathutil.CanonicalPluginName(name)
+				plugin, err := indexscanner.LoadPluginByName(paths.IndexPluginsPath(indexName), pluginName)
 				if err != nil {
 					if os.IsNotExist(err) {
 						return errors.Errorf("plugin %q does not exist in the plugin index", name)
 					}
 					return errors.Wrapf(err, "failed to load plugin %q from the index", name)
 				}
-				install = append(install, plugin)
+				install = append(install, receipt.New(plugin, indexName))
 			}
 
 			if *manifest != "" {
 				plugin, err := indexscanner.ReadPluginFromFile(*manifest)
 				if err != nil {
 					return errors.Wrap(err, "failed to load plugin manifest from file")
 				}
-				install = append(install, plugin)
+				install = append(install, receipt.New(plugin, constants.DefaultIndexName))
```

Comment:
I think we should not convert to `Receipt` until we actually installed, and we're about to write the `Receipt` object to disk. Thoughts @corneliusweig?

This complicates the code a bit, because nothing's installed yet, but scan index dir + convert everything to Receipt in-memory already. Looks odd in the code to me.

That's why I was imagining a small internal type, specific to this pkg that contains Plugin+indexname only.

Reply:
I have something like this in mind:

```go
type pluginEntry {
   p plugin
   indexName string
}
```

Reply:
Ya I was going back and forth on that a lot when I was working on this. It felt weird to add a new type that was essentially the same as `Receipt` but I agree that it is weird/conceptually wrong to create the receipt before the plugin is actually installed.

Reply:
Yeah, this is definitely cleaner. Just because `Receipt` happens to have the field that we want to store in addition does not mean it is the right choice. Consider what happens when we add more fields to `Receipt`. What would happen most likely is that we would not update this instance and then we have a in invalid `Receipt` in our hands. So a dedicated struct is what we need here.